test/lit/passes/unsubtyping.wast

```diff
@@ -109,14 +107,16 @@
 
 (module
  ;; CHECK:      (rec
- ;; CHECK-NEXT:  (type $X (sub (struct)))
+ ;; CHECK-NEXT:  (type $A (sub (struct (field (ref null $X)))))
+
+ ;; CHECK:       (type $X (sub (struct)))
  (type $X (sub (struct)))
+ ;; CHECK:       (type $B (sub $A (struct (field (ref null $Y)))))
+
  ;; CHECK:       (type $Y (sub $X (struct)))
  (type $Y (sub $X (struct)))
 
- ;; CHECK:       (type $A (sub (struct (field (ref null $X)))))
  (type $A (sub (struct (ref null $X))))
```

Comment:
This test becomes less readable with this change, as it moves the check for `$A` away from `$A`. We've worked around this manually in the past, by moving the definition to where the output is, but on a change this large obviously that isn't practical. But just applying this change means making potentially many tests less readable, effectively undoing all the manual effort we've put into these tweaks.

Maybe now is a good time to automate those tweaks, before this PR? I mean that the auto-updater of lit tests could move definitions so that they appear together automatically. Or, it could move the checks if the auto-updater would also accept that change in order.

Reply:
After talking offline, we decided to add an option that propagates the order of types from the input through to the output, similarly to how we propagate type names. Once we can use that option in some of these large tests with many types where the optimized output order doesn't matter, the diff here will become much smaller.